### PR TITLE
Document API guarantees for 1.0.0

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

File: docs/posts/towards-1.0-sdk.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  minor version upgrades.[^1]
+* Any call to such an interface through the published, generated SDKs.
+  These include:
+
+  * [Python `lakefs-sdk`](https://pypi.org/project/lakefs-sdk/)
+  * [Java Maven `io.lakefs:lakefs-sdk`](https://mvnrepository.com/artifact/io.lakefs/lakefs-sdk)
+
+  If your program only uses such calls on the SDK, it will continue to
+  compile and function across a minor version upgrade of the SDK.
+* Any objects created by a lakeFS server version will continue to work after
```
Comment (itaiad200) [inline docs/posts/towards-1.0-sdk.md:64]: Can you define "any objects"?

**Review by itaiad200 — APPROVED** [review]
Thanks for the important document. I think our community should be involved as it effects them the most. Can we share it publicly to a larger audience?

File: docs/posts/towards-1.0-sdk.md (hunk 1 above)
Comment (arielshaqed) [reply docs/posts/towards-1.0-sdk.md]: Done, thanks!

**Comment by arielshaqed** [comment]
@itaiad200 said:
> [...] I think our community should be involved as it effects them the most. Can we share it publicly to a larger audience?

Entirely agreed, I wanted to make sure that we have some agreement on these guarantees on the supply side before asking for comments on the demand side.  SO now I [invited review on #dev](https://lakefs.slack.com/archives/C01APUNTSGH/p1695544004659149).

## ⬇️ ⬇️ ⬇️ ⬇️ Looking forward to seeing comments here!! ⬇️ ⬇️ ⬇️ ⬇️

**Review by Jonathan-Rosenberg — APPROVED** [review]

File: docs/posts/towards-1.0-sdk.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+to 0.105.0 could break old programs.  Indeed that happened.  Starting with
+1.0.0, a minor version that breaks old programs is a bug.
+
+## 1.0.0 is a **major** version upgrade!
+
+All guarantees here are for minor version upgrades.  The upgrade to 1.0.0 is
+of course a **major version upgrade**, and it will necessarily break some
+client programs.
+
+## Interface guarantees
```
Comment (nopcoder) [inline docs/posts/towards-1.0-sdk.md:41]: Do we like to specify that the S3 interface will change based on the compatibility with latest S3 interface?

**Review by nopcoder — APPROVED** [review]

**Review by arielshaqed — COMMENTED** [review]
Thanks!

I'm sending this out to our users :-)

**Pull request merged**